firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java

```diff
@@ -740,8 +740,10 @@ private boolean isInFilter(Target target, FieldPath fieldPath) {
     for (Filter filter : target.getFilters()) {
       if ((filter instanceof FieldFilter) && ((FieldFilter) filter).getField().equals(fieldPath)) {
         FieldFilter.Operator operator = ((FieldFilter) filter).getOperator();
-        return operator.equals(FieldFilter.Operator.IN)
-            || operator.equals(FieldFilter.Operator.NOT_IN);
+        if (operator.equals(FieldFilter.Operator.IN)
```

Comment:
A not-in filter does not have to appear on the first filter for a field. If there are multiple filters for the field path, not in can appear later.